src/AppFile/ZendExpressiveApplicationAdapter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace Cjm\Behat\Psr7Extension\AppFile;
+
+use Cjm\Behat\Psr7Extension\Psr7App;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Expressive\Application;
+
+final class ZendExpressiveApplicationAdapter implements Psr7App
+{
+    /**
+     * @var string
+     */
+    private $path;
+
+    public function __construct(string $path)
+    {
+        if (!file_exists($path)) {
+            throw new Psr7AppFileException('No file found at ' . $path);
+        }
+
+        $this->path = $path;
+    }
+
+    public function handle(RequestInterface $request): ResponseInterface
+    {
+        if (! $request instanceof ServerRequestInterface) {
```

Comment:
This whole thing could be stripped out if the `Psr7App` interface changes to only accept `ServerRequestInterface`, which I think is probably preferable given the intentions...